app/src/main/java/com/duckduckgo/app/browser/omnibar/OmnibarLayout.kt

```diff
@@ -0,0 +1,729 @@
+        when (decoration) {
+            is Mode -> {
+                viewModel.onViewModeChanged(decoration.viewMode)
+            }
+
+            is PrivacyShieldChanged -> {
+                viewModel.onPrivacyShieldChanged(decoration.privacyShield)
+            }
+
+            is Outline -> {
+                viewModel.onOutlineEnabled(decoration.enabled)
+            }
+
+            CancelAnimations -> {
+                cancelTrackersAnimation()
+            }
+
+            is LaunchTrackersAnimation -> {
+                viewModel.onAnimationStarted(decoration)
+            }
+
+            is LaunchCookiesAnimation -> {
+                createCookiesAnimation(decoration.isCosmetic)
+            }
+
+            is ChangeCustomTabTitle -> {
+                updateCustomTabTitle(decoration)
+            }
+
+            is HighlightOmnibarItem -> {
+                viewModel.onHighlightItem(decoration)
+            }
+        }
+    }
+
+    fun reduce(stateChange: StateChange) {
+        if (isAttachedToWindow) {
+            reduceDeferred(stateChange)
+        } else {
+            Timber.d("Omnibar: reduce not attached saving $stateChange")
+            this.stateBuffer.add(stateChange)
+        }
+    }
+
+    private fun reduceDeferred(stateChange: StateChange) {
+        when (stateChange) {
+            is StateChange.LoadingStateChange -> {
+                viewModel.onExternalStateChange(stateChange)
+                onNewProgress(stateChange.loadingViewState.progress, stateChange.onAnimationEnd)
+            }
+
+            else -> {
+                viewModel.onExternalStateChange(stateChange)
+            }
+        }
+    }
+
+    override fun setExpanded(expanded: Boolean) {
+        when (omnibarPosition) {
+            OmnibarPosition.TOP -> super.setExpanded(expanded)
+            OmnibarPosition.BOTTOM -> (behavior as BottomAppBarBehavior).animateToolbarVisibility(
+                expanded,
+            )
+        }
+    }
+
+    override fun setExpanded(
+        expanded: Boolean,
+        animate: Boolean,
+    ) {
+        when (omnibarPosition) {
+            OmnibarPosition.TOP -> super.setExpanded(expanded, animate)
+            OmnibarPosition.BOTTOM -> (behavior as BottomAppBarBehavior).animateToolbarVisibility(
+                expanded,
+            )
+        }
+    }
+
+    override fun getBehavior(): CoordinatorLayout.Behavior<AppBarLayout> {
+        return when (omnibarPosition) {
+            OmnibarPosition.TOP -> TopAppBarBehavior(context, this)
+            OmnibarPosition.BOTTOM -> BottomAppBarBehavior(context, this)
+        }
+    }
+
+    private fun renderScrolling(enabled: Boolean) {
+        if (isAttachedToWindow) {
+            isScrollingEnabled = enabled
+            if (enabled) {
+                omnibarScrolling.enableOmnibarScrolling(toolbarContainer)
+            } else {
+                omnibarScrolling.disableOmnibarScrolling(toolbarContainer)
```

Comment:
This causes the gap issue that was fixed in [this commit](https://github.com/duckduckgo/Android/pull/5131/commits/d2d61ed937b23b91fcefee479c7589976dad3b43).